### PR TITLE
Class Associations

### PR DESCRIPTION
**Pull request by levy002**
In this PR, I:
- [x] created a class Classroom.
- [x] created the has-many/belongs-to relationship between Classroom and Student.
- [x] created a class Book.
- [x] create a class Rental.
- [x] created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

Cheers!!

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 require './person'
 
 class Student < Person
-  def initialize(age, classroom, name, parent_permission: true)
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission)
-    @classroom = classroom
+    classroom.add_student(self) unless classroom.includes?(self)
   end
```
Comment (ShadyShawkat) [inline student.rb]: - [x] [OPTIONAL] Kindly add an `attr_reader` of the classroom instance variable in the `Student` class.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+    @person = person
+    person.add_rental(self)
```
Comment (ShadyShawkat) [inline rental.rb:7]: - [x] Please make sure to implement the `add_rental` function in the `Person` class, as it currently doesn't exist.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @levy002, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -26,4 +26,8 @@ def of_age?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental
```
Comment (danmainah) [inline person.rb]: - [x] Kindly ensure the person itself is set as a parameter when creating the new rental to achieve the 'has-many' association.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @levy002 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Github flow was correctly used


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @levy002 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**